Mathlib/Init/Data/Nat/Lemmas.lean

```diff
@@ -69,14 +61,15 @@ def subInduction {P : ℕ → ℕ → Sort u} (H1 : ∀ m, P 0 m) (H2 : ∀ n, P
   | succ n, succ m => H3 _ _ (subInduction H1 H2 H3 n m)
 
 -- Porting note: added `elab_as_elim`
-@[elab_as_elim]
+@[elab_as_elim, deprecated Nat.strongRecOn (since := "2024-07-27")]
 protected theorem strong_induction_on {p : Nat → Prop} (n : Nat)
```

Comment:
most of this PR is just dealing with the fallout of this deprecation, but I don't see the point. I mean, I realize that `strongRecOn` is `Sort`-valued as opposed to `Prop`-valued, but why not just add a docstring to `strong_induction_on` which says: "This is for proving theorems. If you need to construct data, then use `Nat.strongRecOn`."

Then just move `Nat.strong_induction_on` to a different file.

Reply:
Aha. `Nat.(cases_)strong_induction_on` should still be deprecated, but to `Nat.(cases)strongInductionOn` in `Init.WF`, which I have done. Besides, why would I want to keep a theorem that is a **strictly less general** version of another one and is therefore redundant?

Reply:
I am of the opinion that `Nat.strong_induction_on` should be the Prop-valued specialisation of `Nat.strongRecOn`

Reply:
@YaelDillies do you want me to add the Prop-valued specialisations to my core PR?

Reply:
I think it's fine to have `strong_induction_on` in mathlib only. If people want to upstream it, they always can